server/server/src/billing/payment_service.rs

```diff
@@ -0,0 +1,247 @@
+use crate::billing::stripe_client;
+use crate::file_index_repo::{GetLastStripeCreditCardInfoError, FREE_TIER_SIZE, PAID_TIER_SIZE};
+use crate::ServerError::{ClientError, InternalError};
+use crate::{file_index_repo, RequestContext, ServerError, ServerState};
+use libsecp256k1::PublicKey;
+use lockbook_models::api::{
+    AccountTier, CardChoice, GetLastRegisteredCreditCardError, GetLastRegisteredCreditCardRequest,
+    GetLastRegisteredCreditCardResponse, SwitchAccountTierError, SwitchAccountTierRequest,
+    SwitchAccountTierResponse,
+};
+use sqlx::{Postgres, Transaction};
+
+pub async fn switch_account_tier(
+    context: RequestContext<'_, SwitchAccountTierRequest>,
+) -> Result<SwitchAccountTierResponse, ServerError<SwitchAccountTierError>> {
+    let (request, server_state) = (&context.request, context.server_state);
+
+    let mut transaction = match server_state.index_db_client.begin().await {
+        Ok(t) => t,
+        Err(e) => {
+            return Err(InternalError(format!("Cannot begin transaction: {:?}", e)));
+        }
+    };
+
+    let data_cap = file_index_repo::get_account_data_cap(&mut transaction, &context.public_key)
+        .await
+        .map_err(|e| InternalError(format!("Cannot get account data cap in Postgres: {:?}", e)))?;
+
+    match (data_cap as i64, &request.account_tier) {
+        (FREE_TIER_SIZE, AccountTier::Monthly(card)) => {
+            create_subscription(&context.public_key, server_state, &mut transaction, card).await?;
+        }
+        (FREE_TIER_SIZE, AccountTier::Free) | (_, AccountTier::Monthly(_)) => {
+            return Err(ClientError(SwitchAccountTierError::NewTierIsOldTier));
+        }
+        (_, AccountTier::Free) => {
+            let subscription_id = file_index_repo::get_active_stripe_subscription_id(
```

Comment:
I think for now we need to force people to delete files first, I know this is not ideal, but:
+ we can't pay for people's files (especially of unbounded size)
+ we can't decide what to delete